src/highdicom/sr/utils.py

```diff
@@ -192,18 +192,152 @@ def get_coded_modality(item: Dataset) -> str:
+        '1.2.840.10008.5.1.4.1.1.12.2': codes.cid29.Radiofluoroscopy,
+        '1.2.840.10008.5.1.4.1.1.12.2.1': codes.cid29.Radiofluoroscopy,
+        '1.2.840.10008.5.1.4.1.1.13.1.1': codes.cid29.XRayAngiography,
+        '1.2.840.10008.5.1.4.1.1.13.1.2': codes.cid29.DigitalRadiography,
+        '1.2.840.10008.5.1.4.1.1.13.1.3': codes.cid29.Mammography,
+        '1.2.840.10008.5.1.4.1.1.14.1': codes.cid29.IntravascularOpticalCoherenceTomography,  # noqa E501
+        '1.2.840.10008.5.1.4.1.1.14.2': codes.cid29.IntravascularOpticalCoherenceTomography,  # noqa E501
+        '1.2.840.10008.5.1.4.1.1.20': codes.cid29.NuclearMedicine,
+        '1.2.840.10008.5.1.4.1.1.68.1': codes.cid29.OpticalSurfaceScanner,
+        '1.2.840.10008.5.1.4.1.1.68.2': codes.cid29.OpticalSurfaceScanner,
+        '1.2.840.10008.5.1.4.1.1.77.1.1': codes.cid29.Endoscopy,
+        '1.2.840.10008.5.1.4.1.1.77.1.1.1': codes.cid29.Endoscopy,
+        '1.2.840.10008.5.1.4.1.1.77.1.2': codes.cid29.GeneralMicroscopy,
+        '1.2.840.10008.5.1.4.1.1.77.1.2.1': codes.cid29.GeneralMicroscopy,
+        '1.2.840.10008.5.1.4.1.1.77.1.3': codes.cid29.SlideMicroscopy,
+        '1.2.840.10008.5.1.4.1.1.77.1.4': codes.cid29.ExternalCameraPhotography,
+        '1.2.840.10008.5.1.4.1.1.77.1.4.1': codes.cid29.ExternalCameraPhotography,  # noqa E501
+        '1.2.840.10008.5.1.4.1.1.77.1.5.1': codes.cid29.OphthalmicPhotography,
+        '1.2.840.10008.5.1.4.1.1.77.1.5.2': codes.cid29.OphthalmicPhotography,
+        '1.2.840.10008.5.1.4.1.1.77.1.5.4': codes.cid29.OphthalmicTomography,
+        '1.2.840.10008.5.1.4.1.1.77.1.6': codes.cid29.SlideMicroscopy,
+        '1.2.840.10008.5.1.4.1.1.78.1': codes.cid29.Lensometry,
+        '1.2.840.10008.5.1.4.1.1.78.2': codes.cid29.Autorefraction,
+        '1.2.840.10008.5.1.4.1.1.78.3': codes.cid29.Keratometry,
+        '1.2.840.10008.5.1.4.1.1.78.4': codes.cid29.SubjectiveRefraction,
+        '1.2.840.10008.5.1.4.1.1.78.5': codes.cid29.VisualAcuity,
+        '1.2.840.10008.5.1.4.1.1.78.7': codes.cid29.OphthalmicAxialMeasurements,
+        '1.2.840.10008.5.1.4.1.1.78.8': codes.cid29.Lensometry,
+        '1.2.840.10008.5.1.4.1.1.80.1': codes.cid29.OphthalmicVisualField,
+        '1.2.840.10008.5.1.4.1.1.81.1': codes.cid29.OphthalmicMapping,
+        '1.2.840.10008.5.1.4.1.1.82.1': codes.cid29.OphthalmicMapping,
+        '1.2.840.10008.5.1.4.1.1.128': codes.cid29.PositronEmissionTomography,
+        '1.2.840.10008.5.1.4.1.1.130': codes.cid29.PositronEmissionTomography,
+        '1.2.840.10008.5.1.4.1.1.128.1': codes.cid29.PositronEmissionTomography,
+        '1.2.840.10008.5.1.4.1.1.481.1': codes.cid29.RTImage
     }
-    if item.SOPClassUID in sopclass_to_modalty_map.keys():
-        return sopclass_to_modalty_map[item.SOPClassUID]
+    if sop_class_uid in sopclass_to_modalty_map.keys():
+        return sopclass_to_modalty_map[sop_class_uid]
     else:
         return None
+
+
+def is_dicom_image(sop_class_uid: str) -> bool:
```

Comment:
As mentioned in a previous comment, this should be determined based on attributes that describe the pixel data rather than based on the SOP Class UID.

```suggestion
def is_image(dataset: Dataset) -> bool:
```

Reply:
@seandoyle @hackermd you may want to look at [this method](https://github.com/MGHComputationalPathology/highdicom/blob/8fa057ba15e3dc21270c946dfc25de5c0429cd79/src/highdicom/sr/content.py#L27), which seems very similar in intent

Reply:
Indeed! Thanks for pointing this out @CPBridge.

However, I would suggest changing the existing function ([here](https://github.com/MGHComputationalPathology/highdicom/blob/8fa057ba15e3dc21270c946dfc25de5c0429cd79/src/highdicom/sr/content.py#L50-L54)) and use metadata attributes such as Rows, Columns, etc. rather than the Pixel Data attributes.
In case of large image datasets (whole slide images in pathology) we don't necessary want require users to pass the full data set, but just the metadata.

Reply:
@seandoyle I would either suggest moving the function into the root package (into `src/highdicom/utils.py`), since it may be useful beyond SR (unless this would lead to circular dependencies).

Reply:
Turned out this would lead to a circular dependency (https://github.com/MGHComputationalPathology/highdicom/pull/83#discussion_r754608527).